courses/models.py

```diff
@@ -154,7 +155,7 @@ def num_courses(self):
         """Gets the number of courses in this program"""
         return self.courses.count()
 
-    @property
+    @cached_property
```

Comment:
You will need to update the tests as now the runtime changes to `course.runs` won't be reflected in the `next_run_date` during tests unless explicitly refreshed

Reply:
tests fixed.